docs/overview/architecture.md

```diff
@@ -0,0 +1,240 @@
+
+It will perform the following action:
+
+1. Return error if the user has already input the same vector in Vald
+1. Generate the corresponding UUID for internal use.
+1. Forward the metadata (vector ID and UUID) request to the Vald Meta Agent.
+1. Forward the vector information (vector ID, vector, and UUID) to Vald Backup Gateway.
+
+#### Vald Meta
+
+Vald Meta is the agent to process the CRUD request of the metadata (vector ID and UUID).
+Users can configure which data source to be used in Vald Meta (for example Redis or Cassandra).
+
+### Vald Backup
+
+To support auto-healing functionality and increase performance during disaster recovery, Vald implements the backup mechanism.
+
+#### Vald Compressor
+
+Vald Compressor compresses all of the data (metadata and the vector data) and sends to the Vald Backup Manager to process the backup request.
+
+#### Vald Backup Manager
+
+Vald Backup Manager processes the CRD request of the backup request and handles the compressed metadata. Users can configure which data source to be used in Vald Meta (for example Redis or Cassandra).
+
+#### Vald Backup Gateway
+
+Vald Backup Gateway will forward the backup request to the Vald LB Gateway.
+It also forwards to Vald Compressor asynchronously with metadata.
+
+### Vald Load Balancing
+
+Load balancing is one of the important concepts in distributed computing, which means it distributes a set of tasks over a set of resources aiming for making the overall processing more efficient.
+Vald implements its own load balancing controller.
+Vald can load balance the request base on node resources.
+
+#### Vald LB Gateway
+
+Vald LB Gateway loads balance the user request base on the node resources results from the Agent Discoverer.
+
+#### Agent Discoverer
+
+Agent Discoverer discovers active Vald pods and the corresponding node's resources usage via [kube-apiserver](https://github.com/kubernetes/kubernetes/tree/master/cmd/kube-apiserver).
+
+### Vald Core Engine
+
+In this section, we will describe what is Vald Agent and the corresponding components to support Vald Agent.
+
+#### Vald Agent
+
+Vald Agent is the core of the Vald. By default Vald uses [NGT](https://github.com/yahoojapan/NGT) to provide API for users to insert/update/delete/search vectors.
+
+Each Vald Agent pod holds different high dimensional vector data space, which is constructed by insert/update vectors for searching approximate vectors.
+
+When you request searching with your vector in Vald, each Vald Agent returns different _k_-nearest neighbors' vectors which are similar to the searching vector.
+
+<img src="../../assets/docs/vector_data_space_explain.svg" />
+
+The same vector will be inserted into multiple Vald Agents, not all Vald Agents.
+It makes that the vector data space constructed in each Vald Agent will be different, causing the searching results from each Vald Agent are different.
```

Comment:
![badge](https://img.shields.io/badge/review-suggestion-blue.svg?style=flat-square)

```suggestion
Each Vald Agent pod has own vector data space because only several pods are selected to be inserted every single vector.
When you send a search request to Vald, each Vald Agent return different results of _k_-nearest neighbors depends on their index, and you'll get the merged result of them.

<img src="../../assets/docs/vector_data_space_explain.svg" />
```

Reply:
i'm not sure it is valid syntax, but it can be more simpler than before.

Reply:
I think it is easier to understand, thank you for your suggestion :)
just corrected some wordings, please review :).
```suggestion
Each Vald Agent pod has its own vector data space because only several Vald Agents are selected to be inserted/updated in a single insert/update request.

When searching a vector in Vald, each Vald Agent return different results of _k_-nearest neighbors depending on their index, and you'll get the merged result of them.
<img src="../../assets/docs/vector_data_space_explain.svg" />
```

Reply:
Great! :+1: